File: Arcade/Intro/JavaScript/37_arrayMaxConsecutiveSum.js
function arrayMaxConsecutiveSum(inputArray, k) {
    var max = inputArray.slice(0,k).reduce(sum),
        tmp = max;
    for (var i = k; i < inputArray.length; i++) {
        tmp += inputArray[i] - inputArray[i-k];
        max = tmp > max ? tmp : max;
    }
    return max;
}

function sum(n1,n2) { return n1 + n2; }

File: Arcade/Intro/C++/37_arrayMaxConsecutiveSum.cpp
int arrayMaxConsecutiveSum(std::vector<int> inputArray, int k) {
    int sum = 0;
    for(int i=0;i<k;i++){
        sum+=inputArray[i];
    }
    int max = sum;
    for(int i=1;i<=inputArray.size()-k;i++){
        sum-=inputArray[i-1];
        sum+=inputArray[(i+k)-1];
        if(sum>max) max=sum;
    }
    return max;
}
